\begin{ThesisAbstract}

	\begin{FrenchAbstract}
	
	
			\KeyWords{Régulation génétique, Facteur de transcription, Modèle de
         Potts, Phylogénétique, Algorithme bayésien, différenciation musculaire, trichomes.}
	\end{FrenchAbstract}
	
   \begin{EnglishAbstract}

      Cellular differentiation and tissue specification depend in part on the
      establishment of specific transcriptional programs of gene expression.
      These programs result from the interpretation of genomic regulatory
      information by sequence-specific transcription factors (TFs). Decoding
      this information in sequenced genomes is a key issue.
      First, we present models that describe the interaction between the TFs
      and the DNA sequences they bind to, called Transcription Factor Binding
      Sites (TFBSs). Using a Potts model inspired from spin glass physics along
      with high-throughput binding data for a variety of Drosophilae and
      mammals TFs, we show that TFBSs exhibit correlations among nucleotides
      and that the account of their contribution in the binding energy greatly
      improves the predictability of genomic TFBSs.
      Then, we present a Bayesian, phylogeny-based algorithm designed to
      computationally identify the Cis-Regulatory Modules (CRMs) that control
      gene expression in a set of co-regulated genes. Starting with a small
      number of CRMs in a reference species as a training set, but with no a
      priori knowledge of the factors acting in trans, the algorithm uses the
      over-representation and conservation of TFBSs among related species to
      predict putative regulatory elements along with genomic CRMs underlying
      co-regulation. We show several applications of this algorithm both in
      Drosophila and vertebrates. We also present an extension of the algorithm
      to the case of pattern recognition, showing that CRMs with different
      patterns of expression can be distinguished on the sole basis of their
      DNA motifs content.
	
         \KeyWords{Gene regulation, Transcription Factor, Potts Model,
         Phylogeny, Bayesian algorithm, muscle differentiation, trichomes.}
      \end{EnglishAbstract}
	
\end{ThesisAbstract}
